tkg/managementcomponents/helper.go

```diff
@@ -129,6 +133,27 @@ func convertToString(config interface{}) string {
 	return ""
 }
 
+// convertToNodeNetworkList converts config into avi node network list type, and return default value if config is not set
+func convertToNodeNetworkList(userProviderConfigValues map[string]interface{}) ([]NodeNetwork, error) {
+	var nodeNetworkList []NodeNetwork
+	config := userProviderConfigValues[constants.ConfigVariableAviIngressNodeNetworkList]
+	if config != nil {
+		if err := yaml.Unmarshal([]byte(config.(string)), &nodeNetworkList); err != nil {
+			return nodeNetworkList, errors.Errorf("Invalid node network list %s", config.(string))
+		}
+	} else {
+		// return vsphere network if node network list is not set
+		network_pathes := strings.Split(convertToString(userProviderConfigValues[constants.ConfigVariableVsphereNetwork]), "/")
+		network := network_pathes[len(network_pathes)-1]
+		nodeNetworkList = []NodeNetwork{
+			NodeNetwork{
+				NetworkName: network,
+			},
+		}
+	}
+	return nodeNetworkList, nil
```

Comment:
nit: could we move this return line to line 153 above?
```go
else {
		// return vsphere network if node network list is not set
		network_pathes := strings.Split(convertToString(userProviderConfigValues[constants.ConfigVariableVsphereNetwork]), "/")
		network := network_pathes[len(network_pathes)-1]
		nodeNetworkList = []NodeNetwork{
			NodeNetwork{
				NetworkName: network,
			},
		}
	return nodeNetworkList, nil
	}
}
```

Reply:
there will be error if we don't have one return line outside of `if else`